test/models/announcement_test.exs

```diff
@@ -1,34 +1,89 @@
 defmodule Constable.AnnouncementTest do
   use Constable.ModelCase, async: true
   alias Constable.Announcement
+  alias Ecto.Changeset
 
   test "inserting a record sets the last_discussed_at" do
     announcement = build(:announcement) |> insert
 
     assert announcement.last_discussed_at
   end
 
-  test "last_discussed_first" do
-    oldest = insert(:announcement, last_discussed_at: Constable.Time.days_ago(1))
-    newest = insert(:announcement, last_discussed_at: Constable.Time.now)
+  describe ".update_changeset/2" do
+    test "generates a slug from the new title" do
+      title = "A normal title"
+      announcement = insert(:announcement)
 
-    announcements = Announcement.last_discussed_first |> Repo.all
+      slug =
+        announcement
+        |> Announcement.update_changeset(%{title: title})
+        |> Changeset.get_change(:slug)
 
-    assert List.first(announcements).id == newest.id
-    assert List.last(announcements).id == oldest.id
+      assert slug == "a-normal-title"
+    end
+
+    test "does not generate slug if not title is nil" do
```

Comment:
I think this has an extra "not" in it:

"does not generate slug if title is nil"

Reply:
Updated in 3fac16c8443a6fd0449f449569ad8b716a5d86cd